packages/core/admin/ee/server/services/audit-logs.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const createAuditLogsService = (strapi) => {
+  const saveEvent = (name, ...args) => {
+    if (!name) {
+      throw Error('Name is required');
+    }
+    // TODO: filter events here
+    // TODO: save events here via provider
+    console.log(`Listened to event ${name} with args: ${JSON.stringify(args)}`);
+  };
+
+  return {
+    register() {
+      this.unsubscribe = strapi.eventHub.subscribe(saveEvent);
```

Comment:
I'm not sure I understand, the code reads very strangely `this.unsubscribe = subscribe`?

Reply:
The subscribe method returns a function that, when called, will undo the subscribe. So here we are storing this on the audit logs object to easily have access to it later in the destroy

Alex told us it was a common pattern to have cleanup functions like this. I think the most well known example is [useEffect](https://beta.reactjs.org/apis/react/useEffect#connecting-to-an-external-system) in react

Reply:
Oh ok yeah forgot it returned that cleanup function. Thanks for the explanation.